app/mailers/startup_mailer.rb

```diff
@@ -1,18 +1,17 @@
 # Mails sent out to teams, as a whole.
 class StartupMailer < SchoolMailer
   def feedback_as_email(startup_feedback)
-    emails_fullnames = startup_feedback.timeline_event.founders.map { |e| {"#{e.fullname} <#{e.email}>" => {"fullname" => e.fullname, "email" => e.email}} }
-    emails_fullnames.each do | student |
-        student.each do |send_to, values|
-          send_single_email(send_to, values, startup_feedback)
-        end
+    emails_fullnames = startup_feedback.timeline_event.founders.map { |e| [e.fullname, e.email] }
+    emails_fullnames.each do | fullname, email |
```

Comment:
could you turn your `map` into a `do...end` block and then you'd save one more loop

Reply:
```ruby
startup_feedback.timeline_event.founders.map do |founder|
  fullname = founder.fullname
  email = founder.email
  send_to = "#{fullname} <#{email}>"
  send_single_email(send_to, fullname, email, startup_feedback)
end
```

Reply:
@bodhish awesome thanks for sharing. 

```ruby
Hello <%= @students %>, 
```

For this line, in the event students are on a team but they submit their work individually, would the emails still be addressed to every student? I think in this case it should only be addressed to the student who submitted the work.

I also think it should address them just by their first name or first and middle name if they have a middle name rather than their full name. Their full name will feel colder, distant, and more impersonal.

Reply:
> For this line, in the event students are on a team but they submit their work individually, would the emails still be addressed to every student? I think in this case it should only be addressed to the student who submitted the work.

The email will only be fired to students who are part of the submission.

> I also think it should address them just by their first name or first and middle name if they have a middle name rather than their full name. Their full name will feel colder, distant, and more impersonal.

We are just storing the `fullname` for a user, so I don't think it would be possible.

Reply:
> We are just storing the name for a user, so I don't think it would be possible. (won't be straightforward)

What about:

```ruby
def first_name
  name.split(" ", 2).first
end

def last_name
  name.split(" ", 2).last
end
```

This code will split the name at the first space that is present. Therefore double and middle names will show up in the `first_name` method but that's ok I would think.

Reply:
@harigopal thanks for sharing. I read the article. While it shares the idea of a "Preferred name" field at the end, it actually recommends not doing that from a UX perspective.

Getting to a perfect solution here might be intractable. That said, from my limited research, it seems like most western countries, much of India, China, etc. have the concept of first names or personal names.

And I have a feeling that over time more and more cultures will adopt a global naming convention that fits well enough within this pattern.

In aggregate, if most students do have a first name, having the in app and email communications refer to them by their first name, I feel will lead students, again in aggregate, to have a better user experience.

I know if I use software that refers to me everywhere by full name, it feels like a UX smell and I'm less inclined to want to adopt or trust the tool. It's probably irrational but I bet it's not just me that has this feeling.

This sort of change would come with a trade-off for sure. Food for thought. Reflect on it and see if you can see the trade-off being worth it.